examples/sync/src/bin/server.rs

```diff
@@ -0,0 +1,438 @@
+where
+    E: commonware_runtime::Storage
+        + commonware_runtime::Clock
+        + commonware_runtime::Metrics
+        + commonware_runtime::Network,
+{
+    info!(client_addr = %client_addr, "Client connected");
+
+    loop {
+        // Read length-prefixed message
+        let message_data = match read_message(&mut stream).await {
+            Ok(data) => data,
+            Err(NetworkError::ReadFailed(_)) => {
+                info!(client_addr = %client_addr, "Client disconnected");
+                break;
+            }
+            Err(e) => {
+                error!(client_addr = %client_addr, error = %e, "❌ Connection error");
+                state.inc_errors();
+                break;
+            }
+        };
+
+        // Parse the message
+        let message: Message = match Message::decode(&message_data[..]) {
+            Ok(msg) => msg,
+            Err(e) => {
+                error!(client_addr = %client_addr, error = %e, "❌ Failed to parse message");
+                state.inc_errors();
+                continue;
+            }
+        };
+
+        // Handle the message
+        let response = match message {
+            Message::GetOperationsRequest(request) => {
+                match handle_get_operations_request(&state, request).await {
+                    Ok(response) => Message::GetOperationsResponse(response),
+                    Err(e) => {
+                        warn!(client_addr = %client_addr, error = %e, "❌ GetOperations failed");
+                        state.inc_errors();
+                        Message::Error(e.into())
+                    }
+                }
+            }
+            Message::GetServerMetadataRequest(request) => {
+                match handle_get_server_metadata_request(&state, request).await {
+                    Ok(response) => Message::GetServerMetadataResponse(response),
+                    Err(e) => {
+                        warn!(client_addr = %client_addr, error = %e, "❌ GetServerMetadata failed");
+                        state.inc_errors();
+                        Message::Error(e.into())
+                    }
+                }
+            }
+            _ => {
+                warn!(client_addr = %client_addr, "❌ Unexpected message type");
+                state.inc_errors();
+                Message::Error(ErrorResponse::new(
+                    None,
+                    commonware_sync::ErrorCode::InvalidRequest,
+                    "Unexpected message type".to_string(),
+                ))
+            }
+        };
+
+        // Send the response with length prefix
+        let response_data = response.encode().to_vec();
+        if let Err(e) = send_message(&mut sink, &response_data).await {
+            error!(client_addr = %client_addr, error = %e, "❌ Failed to send response");
+            state.inc_errors();
+            break;
+        }
+    }
+
+    Ok(())
+}
+
+fn main() {
+    // Initialize tracing with a clean format
```

Comment:
In other binaries, I use `tokio::telemetry` for all of this: https://github.com/commonwarexyz/alto/blob/3bcb093342b32fd1c19733516419d905fc5e67eb/chain/src/bin/validator.rs#L80-L93

This also allows us to query either binary for metrics (although it may mess with your log formatting). We should probably remove this line: https://github.com/commonwarexyz/monorepo/blob/58aa05b738986567b43507243ef2451feca27eab/runtime/src/tokio/telemetry.rs#L53

Reply:
For an example like this, ok to leave but figured I'd call out for your knowledge.

Reply:
> We should probably remove this line (makes it very spacy)

Do you want me to make an issue/PR for this or leave for now? I'm not especially opinionated. Perhaps we could offer more options, instead of just JSON vs not-JSON (i.e. JSON vs pretty)? I see there is also a `compact()` log option available.